server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstraction.java

```diff
@@ -235,6 +235,7 @@ public Alias(AliasMetadata aliasMetadata, List<IndexMetadata> indexMetadatas) {
                 }
                 isSystem = isSystem && imd.isSystem();
             }
+            this.referenceIndices.sort(Index.COMPARE_BY_NAME);
```

Comment:
Needed this so the equals check in the assertion on the index lookup becomes stable. I think the performance overhead of this is rather small and the savings from this change make up for it easily I believe.